src/HeapJobOrderer.ts

```diff
@@ -0,0 +1,156 @@
+import { Heap } from 'heap-js'
+import { Job } from './Job'
+import { JobOrderer } from './JobOrderer'
+import assert = require('assert')
+
+/**
+ * Manages a set of jobs that have to be run.
+ */
+export class HeapJobOrderer implements JobOrderer {
+	private sourcesHeap: Heap<Job>
+
+	private nonSources: Set<Job> = new Set()
+
+	private inProgress: Set<Job> = new Set()
+
+	private jobToDependents: Map<Job, Set<Job>> = new Map<Job, Set<Job>>()
+
+	private jobToCompletedPrereqs: Map<Job, Set<Job>> = new Map<Job, Set<Job>>()
+
+	/**
+	 * Create the JobOrderer and inform it of the jobs to manage.
+	 *
+	 * @throws When one of the jobs has a prerequisite that is not one of the jobs, or is itself.
+	 * @param jobs - The jobs to manage. They must have correctly populated prerequisites fields.
+	 */
+	constructor(rootJobs: Job[]) {
+		// Initialize the sources heap with its sorting function.
```

Comment:
In this comment, describe what the sorting function does (I'm guessing descending by number of dependents?) and why.